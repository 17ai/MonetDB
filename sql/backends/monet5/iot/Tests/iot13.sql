-- Introduce firing conditions to the queries themselves.
set schema iot;
set optimizer='iot_pipe';

create stream table tmp13 (t timestamp, sensor integer, val decimal(8,2)) ;
create table agenda13(t timestamp, cnt integer, msg string);

-- Queries can fire based on both the actual state of the streams and heartbeat
call iot.window('iot','tmp13',4);

-- every 5 seconds inspect the basket regardless filling
call iot.heartbeat('iot','tmp13',5000);

create procedure cq13()
begin
	insert into agenda13 select count(*), 'full batch' from tmp13;
end;

call iot.query('iot','cq13');
call iot.pause();

insert into tmp13 values('2005-09-23 12:34:26.736',1,12.34);
select * from tmp13;

-- reactivate all continuous queries
call iot.resume();
-- wait for 1 cycle in the scheduler
call iot.wait(1);

select 'RESULT';
select * from agenda13;

select * from iot.errors();
call iot.stop();
drop procedure cq13;
drop table tmp13;
drop table agenda13;
