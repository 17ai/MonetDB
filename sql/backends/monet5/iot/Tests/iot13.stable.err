stderr of test 'iot13` in directory 'sql/backends/monet5/iot` itself:


# 11:32:49 >  
# 11:32:49 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=30951" "--set" "mapi_usock=/var/tmp/mtest-13328/.s.monetdb.30951" "--set" "monet_prompt=" "--forcemito" "--dbpath=/export/scratch1/mk/iot//Linux/var/MonetDB/mTests_sql_backends_monet5_iot"
# 11:32:49 >  

# builtin opt 	gdk_dbpath = /export/scratch1/mk/iot//Linux/var/monetdb5/dbfarm/demo
# builtin opt 	gdk_debug = 0
# builtin opt 	gdk_vmtrim = no
# builtin opt 	monet_prompt = >
# builtin opt 	monet_daemon = no
# builtin opt 	mapi_port = 50000
# builtin opt 	mapi_open = false
# builtin opt 	mapi_autosense = false
# builtin opt 	sql_optimizer = default_pipe
# builtin opt 	sql_debug = 0
# cmdline opt 	gdk_nr_threads = 0
# cmdline opt 	mapi_open = true
# cmdline opt 	mapi_port = 30951
# cmdline opt 	mapi_usock = /var/tmp/mtest-13328/.s.monetdb.30951
# cmdline opt 	monet_prompt = 
# cmdline opt 	gdk_dbpath = /export/scratch1/mk/iot//Linux/var/MonetDB/mTests_sql_backends_monet5_iot
# cmdline opt 	gdk_debug = 536870922

# 11:32:49 >  
# 11:32:49 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-13328" "--port=30951"
# 11:32:49 >  


# 11:32:54 >  
# 11:32:54 >  "Done."
# 11:32:54 >  

