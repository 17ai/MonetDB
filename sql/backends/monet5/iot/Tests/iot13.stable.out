stdout of test 'iot13` in directory 'sql/backends/monet5/iot` itself:


# 11:32:49 >  
# 11:32:49 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=30951" "--set" "mapi_usock=/var/tmp/mtest-13328/.s.monetdb.30951" "--set" "monet_prompt=" "--forcemito" "--dbpath=/export/scratch1/mk/iot//Linux/var/MonetDB/mTests_sql_backends_monet5_iot"
# 11:32:49 >  

# MonetDB 5 server v11.24.0
# This is an unreleased version
# Serving database 'mTests_sql_backends_monet5_iot', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 128bit integers
# Found 15.589 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2016 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://vienna.da.cwi.nl:30951/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-13328/.s.monetdb.30951
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded
# MonetDB/iot loaded

Ready.
#Tumbled iot.tmp13[0] 1 elements
#Tumbled iot.tmp13[1] 1 elements
#Tumbled iot.tmp13[2] 1 elements
#Tumbled iot.tmp13[0] 0 elements
#Tumbled iot.tmp13[1] 0 elements
#Tumbled iot.tmp13[2] 0 elements

# 11:32:49 >  
# 11:32:49 >  "mclient" "-lsql" "-ftest" "-Eutf-8" "-i" "-e" "--host=/var/tmp/mtest-13328" "--port=30951"
# 11:32:49 >  

#set schema iot;
#set optimizer='iot_pipe';
#create stream table tmp13 (t timestamp, sensor integer, val decimal(8,2)) ;
#create table agenda13(i integer, msg string);
#create procedure cq13()
#begin
#	insert into agenda13 select count(*), 'full batch' from tmp13;
#end;
#insert into tmp13 values('2005-09-23 12:34:26.736',1,12.34);
[ 1	]
#insert into tmp13 values('2005-09-23 12:34:26.736',1,12.35);
[ 1	]
#insert into tmp13 values('2005-09-23 12:34:26.736',1,12.36);
[ 1	]
#select 'RESULT';
% .L2 # table_name
% L2 # name
% char # type
% 6 # length
[ "RESULT"	]
#select * from agenda13;
% iot.agenda13,	iot.agenda13 # table_name
% i,	msg # name
% int,	clob # type
% 1,	10 # length
[ 2,	"full batch"	]
[ 1,	"full batch"	]
#select * from iot.errors();
% .L1,	.L1 # table_name
% table,	error # name
% clob,	clob # type
% 0,	0 # length
#drop procedure cq13;
#drop table tmp13;
#drop table agenda13;

# 11:32:54 >  
# 11:32:54 >  "Done."
# 11:32:54 >  

